DEV-INTRO.md

```diff
@@ -1,59 +1,76 @@
+The recommended workflow is gitflow which is described in the PEcAn documentation. For the repository we recommend using a [fork of the PEcAn repsitory](https://help.github.com/en/github/getting-started-with-github/fork-a-repo). In the scripts folder there is a script called [syngit.sh](scripts/syncgit.sh) that will help with synchronizing your fork with the official repository.
 
-### Modules (R packages)
+## Developing in Docker
 
-* General
-** all
-** utils
-** db
-* Analysis
-** modules/meta.analysis 
-** modules/uncertainty"
-** modules/data.land 
-** modules/data.atmosphere
-** modules/assim.batch 
-** modules/assim.sequential 
-** modules/priors
-* Model Interfaces
-** models/ed 
-** models/sipnet 
-** models/biocro
+To get started with development in docker we need to bring up the docker stack first. In the main pecan folder you will find the [docker-compose.yml](docker-compose.yml) file that can be used to bring up the pecan stack. There is also the [docker-compose.dev.yaml](docker-compose.dev.yaml) file that adds additional containers, and changes some services to make it easier for development.
 
+You can copy the `docker-compose.dev.yaml` to `docker-compose.override.yml`. Once that is done the `docker-compose` program will automatically use the `docker-compose.yml` and `docker-compose.override.yml` files, and you don't have to explicitly load them in the commands below.
 
-#### List of modules 
+### First time setup
 
-Installing PEcAn
-----------------
+You can copy the [`env.example`](docker/env.example) file as .env in your pecan folder. The variables we want to modify are:
 
-### Virtual Machine
+* `COMPOSE_PROJECT_NAME` set this to pecan, the prefix for all containers
+* `PECAN_VERSION` set this to develop, the docker image we start with
 
-* Fastest way to get started
-* see PEcAn demo ...
+Next we will create the folders that will hold all the data for the docker containers using: `mkdir -p volumes/{lib,pecan,portainer,postgres,rabbitmq,traefik}`. The `volumes` folder will be ignored by git.
 
-### Installing from source
+First we bring up postgresql (we will start RabbitMQ as well since it takes some time to start): `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d postgres rabbitmq`. This will start postgresql and rabbitmq. We need to wait for a few minutes (you can look at the logs using `docker-compose logs postgres`) to see if it is ready.
 
-#### From GitHub
+During the time it takes for the database to start we will copy the R packages from a container to our `volumes/lib` folder. This folder contain the compiled R packages. Later we will put our newly compiled code here as well. You can copy all the data using `docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -r /usr/local/lib/R/site-library/* /rlib/`. This will copy all compiled packages to your local machine. This only needs to be done once (or if the PEcAn base image changes drastically, for example a new version of R).
```

Comment:
As a first timer, I can't figure out what's going on at this step (or why it needs to be done)

Reply:
agreed with Mike, not clear for me as well. 

Why do we physically copy things instead of mounting? as mentioned above I didn't do any of these.. I guess I followed a different approach, though, I did git clone pecan and built the native version, and mounted my source code folder to docker container (per current documentation - [Local development and testing with docker](https://pecanproject.github.io/pecan-documentation/develop/docker-index.html#docker-local-devel)). I guess that's why I have all compiled packages on my local machine and I don't need to do this step? I think my approach defeats the purpose of not dealing with native pecan build, but now that I did it, I think I like my way better (for my purposes, I'm not saying it should be the way described on the documentation). But I'm also curious if what I did has disadvantages?

Reply:
updated documentation to clarify this. If you share your Rlib folder from your laptop and the docker images you might run into a case where the R version on your laptop and the R version in the docker containers differ, and the resulting compiled packages don't work.

Reply:
I just have one more clarifying question, if I am determined to have a native pecan build in my machine at all times, do I need to worry about incompatibility about R version on my machine and the R version in the docker? I mean I would need to make all changes/updates on my local machine to have the native pecan build anyway right?

Reply:
sounds good to me, thank you.

Reply:
as long as you are not sharing the R library folders, this should work. You can think of the docker containers as mini vm's. The best part about what you want to do is if the two versions of R are different you are testing the code at that point with different versions of R.